spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -659,10 +659,7 @@ private Table load(Identifier ident) {
 
       Matcher branch = BRANCH.matcher(ident.name());
       if (branch.matches()) {
-        Snapshot branchSnapshot = table.snapshot(branch.group(1));
-        if (branchSnapshot != null) {
-          return new SparkTable(table, branchSnapshot.snapshotId(), !cacheEnabled);
-        }
+        return new SparkTable(table, branch.group(1), !cacheEnabled);
```

Comment:
Do we want to keep the "branch_" prefix in the identifier? I just removed it in the previous PR and everything after the table part in the identifier is treated as a branch. https://github.com/apache/iceberg/pull/6651/files#diff-6028f80634c027dee9ea947b9f0e50a51b1e5ced1834b902f0f47303514d0e96L660

I'm okay either way one benefit is that from a user experience point of view it's explicit it's a branch and will less likely collide with column identifier. It's just a bit more verbose.

Reply:
Yeah that was confusing to me that's why I did not add it here, I'd like to know what other people thinks for `table.branch_b` vs `table.b`.

Reply:
I think we should use the same in reads and writes. Otherwise, it is confusing.

Reply:
I think we should start with the strict `branch_` names like this uses. The `VERSION AS OF` syntax provides a better way in the long term, so we can be strict for now.

Reply:
Let me resolve this thread as we reached consensus.